tests/it/runtime/dub.d

```diff
@@ -916,3 +916,33 @@ unittest {
         ninja.shouldExecuteOk;
     }
 }
+
+
+@("extra dflags")
+@Tags("dub", "ninja")
+unittest {
+    import std.file: readText;
+    import std.string: replace;
+
+    with(immutable ReggaeSandbox()) {
+        writeFile(
+            "dub.sdl",
+            [
+                `name "foo"`,
+                `targetType "executable"`,
+                `lflags "-some_flag"`,
+            ],
+        );
+        writeFile(
+            "source/app.d",
+            [
+                q{void main() {}},
+            ],
+        );
+
+        runReggae("-b", "ninja", "--dub-build-type=plain", "--dflag=-Xcc=-fuse-ld=lld");
```

Comment:
Shouldn't we have `--lflags` instead?

Reply:
dflags is all we need, as all linker flags can be specified this way too. And some `-flto=thin` then applies to both compile and link steps, exactly what we need. It's only dub with its notion of `lflags` - which are only a subset of linking-specific flags (and I hate it). But I digress...

Reply:
How else would it be done? Compiling and linking are different things, although flags that need to apply to both might seem to complicate things, they don't: pass them in both.

Reply:
I totally disagree. Dub's `dflags` and DFLAGS apply to both compile & link - because they are simple D compiler flags. There's some existing logic to check if a dflag affects linking, in which case it is used for linking steps too - and we're exactly using that one in reggae too. A whole lot of flags make perfect sense for both compilation & linking. Dub's `lflags` are commands for the underlying linker - Windows link.exe or some ld-compatible linker on Posix.